swift/Arrow/Sources/Arrow/ArrowReader.swift

```diff
@@ -0,0 +1,163 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import FlatBuffers
+import Foundation
+
+public enum ArrowError: Error {
+    case runtimeError(String)
+}
+
+let FILEMARKER = "ARROW1"
+let CONTINUATIONMARKER = -1
+
+public class ArrowReader {
```

Comment:
We have "streaming format" https://arrow.apache.org/docs/format/Columnar.html#ipc-streaming-format and "file format" https://arrow.apache.org/docs/format/Columnar.html#ipc-file-format .
It seems that this is a reader for "file format".
We may want to add "file" to this class name such as `ArrowFileReader`.

Reply:
> the file format wraps the stream format with some extra data

Correct.

But the file format can random read:

https://arrow.apache.org/docs/format/Columnar.html#ipc-file-format 

> This enables random access to any record batch in the file.

The streaming format needs to read from start to end. Because delta dictionary replacement is supported in the streaming format.

See also: https://arrow.apache.org/docs/format/Columnar.html#ipc-file-format 

> In the file format, there is no requirement that dictionary keys should be defined in a DictionaryBatch before they are used in a RecordBatch, as long as the keys are defined somewhere in the file. Further more, it is invalid to have more than one non-delta dictionary batch per dictionary ID (i.e. dictionary replacement is not supported). Delta dictionaries are applied in the order they appear in the file footer.

Reply:
> This class contains a method for fromFile and fromStream.

I see. I misread the implementation. Sorry. I think that it's OK with the current implementation as the first step. We can improve it in follow-up pull requests.